sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/PartitioningAwareFileCatalog.scala

```diff
@@ -96,7 +96,10 @@ abstract class PartitioningAwareFileCatalog(
 
   protected def inferPartitioning(): PartitionSpec = {
     // We use leaf dirs containing data files to discover the schema.
-    val leafDirs = leafDirToChildrenFiles.keys.toSeq
+    val leafDirs = leafDirToChildrenFiles.filterNot {
+      // SPARK-15895: Metadata files like Parquet summary files should not be counted as data files.
+      case (_, files) => files.forall(_.getPath.getName.startsWith("_"))
```

Comment:
For compatibility with https://github.com/rxin/spark/blob/705a76f84605f1d54e98caae7f81631bf80c8feb/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/fileSourceInterfaces.scala#L347 would it be helpful to also exclude any `.`-prefixed files?